compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -817,7 +817,7 @@ public void visit(BLangInvocation iExpr) {
     @Override
     public void visit(BLangErrorConstructorExpr errorConstructorExpr) {
         rewriteExprs(errorConstructorExpr.positionalArgs);
-        rewriteExprs(errorConstructorExpr.namedArgs);
+        errorConstructorExpr.errorDetail.accept(this);
```

Comment:
Instead of directly calling accept, we should do

```suggestion
        rewriteExpr(errorConstructorExpr.errorDetail);
```

This does accept + some other things.